File: Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/model/invocation/AnyInCodeSystemInvocation.java
package org.cqframework.cql.cql2elm.model.invocation;

import java.util.ArrayList;
import java.util.List;
import org.hl7.elm.r1.AnyInCodeSystem;
import org.hl7.elm.r1.Expression;

/**
 * Created by Bryn on 9/12/2018.
 */
public class AnyInCodeSystemInvocation extends OperatorExpressionInvocation<AnyInCodeSystem> {
    public AnyInCodeSystemInvocation(AnyInCodeSystem expression) {
        super(expression);
    }

    @Override
    public List<Expression> getOperands() {
        List<Expression> result = new ArrayList<>();
        result.add(expression.getCodes());
        if (expression.getCodesystemExpression() != null) {
            result.add(expression.getCodesystemExpression());
        }
        return result;
    }

    @Override
    public void setOperands(List<Expression> operands) {
        if (operands == null || operands.isEmpty() || operands.size() > 2) {
            throw new IllegalArgumentException("AnyInCodeSystem operation requires one or two operands.");
        }

        expression.setCodes(operands.get(0));
        if (operands.size() > 1) {
            expression.setCodesystemExpression(operands.get(1));
        }
    }
}

File: cql-to-elm/src/commonMain/kotlin/org/cqframework/cql/cql2elm/model/invocation/AnyInCodeSystemInvocation.kt
package org.cqframework.cql.cql2elm.model.invocation

import org.hl7.elm.r1.AnyInCodeSystem
import org.hl7.elm.r1.Expression

/** Created by Bryn on 9/12/2018. */
class AnyInCodeSystemInvocation(expression: AnyInCodeSystem) :
    OperatorExpressionInvocation<AnyInCodeSystem>(expression) {
    override var operands: List<Expression>
        get() = listOfNotNull(expression.codes, expression.codesystemExpression)
        set(operands) {
            require(operands.size in 1..2) {
                "AnyInCodeSystem operator requires one or two operands."
            }
            expression.codes = operands[0]
            if (operands.size > 1) {
                expression.codesystemExpression = operands[1]
            }
        }
}
